### PR TITLE
 Fix race condition on blockWriteStream

### PR DESCRIPTION
**Pull request by yjh0502**
Fix race condition on blockWriteStream.packets. Use channel instead to
 slice to check ack messages. It fixes race condition on packets and
 gives proper rate controls with channel size.

### CODE REVIEW

File: rpc/block_write_stream.go [hunk 1]
```diff
@@ -115,8 +118,6 @@ func (s *blockWriteStream) finish() error {
 	if err != nil {
 		return err
 	}
-
-	// The last packet has no data; it's just a marker that the block is finished.
```
Comment (colinmarc) [inline rpc/block_write_stream.go]: Any reason for removing this comment?

**Comment by colinmarc** [comment]
Hey! Thanks for the contribution!

This looks good - can you also add a test in another PR that fails without this code? If it ends up being too gnarly, that's fine.

File: rpc/block_write_stream.go (hunk 1 above)
Comment (yjh0502) [inline rpc/block_write_stream.go]: The comment is accidentally deleted, I just submitted a fix.

**Comment by yjh0502** [comment]
Thanks for a review. The problem only appears on highly loaded system, so I'm not sure if I can write a testcase that reproduces the problem.

**Comment by colinmarc** [comment]
Ok, that makes sense. Thanks again!

**Pull request merged**